homeassistant/components/ddwrt/device_tracker.py

```diff
@@ -140,26 +145,31 @@ def _update_info(self):
 
         return True
 
-    def get_ddwrt_data(self, url):
+    async def get_ddwrt_data(self, url):
         """Retrieve data from DD-WRT and return parsed result."""
         try:
-            response = requests.get(
+            response = await self.websession.get(
```

Comment:
The device interface code and data parsing needs to be extracted to a standalone library published on pypi.

https://developers.home-assistant.io/docs/creating_platform_code_review#5-communication-with-devicesservices